### PR TITLE
Add flow rate feature

### PR DESCRIPTION
**Pull request by carlosmolinet**
**Summary**
This PR introduces a flow_rate property to the aioacaia library, enabling the calculation of real-time flow rates based on weight history. This enhancement is a foundational step towards implementing a brew by weight feature, which can be particularly useful for certain espresso machines, such as the La Marzocco Micra.

**Proposed Solution**
The new flow_rate property calculates flow rates in grams per second using the following approach:

- Utilizes weight measurements recorded at least three readings ago for accuracy.
- Disregards data points older than 5 seconds to maintain relevance.
- Validates flow rate values, discarding unrealistic rates exceeding 20 g/s.
- Applies an Exponential Moving Average (EMA) to smooth variations and provide a stable, reliable flow rate estimate.

### CODE REVIEW

File: aioacaia/acaiascale.py [hunk 1]
```diff
@@ -138,6 +142,45 @@ def timer(self) -> int:
             return 0
 
         return int(self._timer_stop - self._timer_start)
+    
+    @property
+    def flow_rate(self) -> float:
```
Comment (zweckj) [inline aioacaia/acaiascale.py]:
```suggestion
    def flow_rate(self) -> float | None:

```

File: aioacaia/acaiascale.py [hunk 2]
```diff
@@ -94,6 +95,9 @@ def __init__(
         self._device_state: AcaiaDeviceState | None = None
         self._weight: float | None = None
 
+        # flow rate
+        self.weight_history = deque(maxlen=20)  # Limit to 20 entries
```
Comment (zweckj) [inline aioacaia/acaiascale.py]:
```suggestion
        self.weight_history: deque[tuple[float, float]] = deque(maxlen=20)  # Limit to 20 entries

```

File: aioacaia/acaiascale.py [hunk 3]
```diff
@@ -138,6 +142,45 @@ def timer(self) -> int:
...
+        for i in range(3, len(self.weight_history)):
+            prev_time, prev_weight = self.weight_history[i - 3]
+            curr_time, curr_weight = self.weight_history[i]
+
+            time_diff = curr_time - prev_time
+            weight_diff = curr_weight - prev_weight
+
+            # Ignore weights older than 5 seconds
+            if now - prev_time > 5:
+                continue
```
Comment (zweckj) [inline aioacaia/acaiascale.py]: this I don't understand tbh. Shouldn't we just drop the items from the list then?

**Review by zweckj — CHANGES_REQUESTED** [review]

File: aioacaia/acaiascale.py (hunk 3 above)
Comment (carlosmolinet) [reply aioacaia/acaiascale.py]: I adjusted this to remove entries older than 5 seconds before the loop begins.

**Review by zweckj — APPROVED** [review]

**Comment by zweckj** [comment]
Thanks 👍 @carlosmolinet

**Pull request merged**